.github/workflows/build.yml

```diff
@@ -0,0 +1,70 @@
+name: Build
+
+on: [push, pull_request]
+
+jobs:
+  build_idf_master:
+    name: build-with-IDF-master
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout esp-protocols
+        uses: actions/checkout@master
+        with:
+          path: esp-protocols
+      - name: Checkout IDF
+        uses: actions/checkout@master
+        with:
+          repository:  espressif/esp-idf
+          path: esp-idf
+
+      - name: Build-examples-with-IDF
+        run: |
+          cd $GITHUB_WORKSPACE/esp-idf
+          git submodule update --init --recursive
+          ./install.sh
+          source export.sh
+          cd $GITHUB_WORKSPACE/esp-protocols/components/esp_modem/examples/simple_cmux_client
+          idf.py build
+          cd $GITHUB_WORKSPACE/esp-protocols/components/esp_modem/examples/ap_to_pppos
+          idf.py build
+          cd $GITHUB_WORKSPACE/esp-protocols/components/esp_modem/examples/modem_console
+          idf.py build
+          cd $GITHUB_WORKSPACE/esp-protocols/components/esp_modem/examples/pppos_client
+          idf.py build
+
+  build:
+    strategy:
+      matrix:
+        idf_ver: ["v4.1", "v4.2", "v4.3", "v4.4"]
+        examples: ["pppos_client", "modem_console", "ap_to_pppos", "simple_cmux_client"]
+        idf_target: ["esp32"]        
+        exclude:
+          - idf_ver: "v4.1"
+            examples: modem_console
+          - idf_ver: "v4.1"
+            examples: ap_to_pppos
+          - idf_ver: "v4.1"
+            examples: simple_cmux_client
+          - idf_ver: "v4.2"
+            examples: simple_cmux_client
+
+    runs-on: ubuntu-20.04
+    container: espressif/idf:release-${{ matrix.idf_ver }}
+    steps:
+      - name: Checkout IDF
+        uses: actions/checkout@v1
+        with:
+          submodules: recursive
+      - name: Checkout esp-protocols
+        uses: actions/checkout@master
+        with:
+          path: esp-protocols
+      - name: Build for ESP32
```

Comment:
You can use `Build for ${{ matrix.idf_target }}` here — i should fix this in libnmea-esp32 as well...